src/librustdoc/html/render/span_map.rs

```diff
@@ -231,6 +231,24 @@ impl<'tcx> Visitor<'tcx> for SpanMapVisitor<'tcx> {
         self.handle_pat(p);
     }
 
+    fn visit_qpath(&mut self, qpath: &QPath<'tcx>, id: HirId, span: Span) {
```

Comment:
You should now be able to remove the `visit_pat` and `handle_pat` methods entirely thanks to this overwrite.

Reply:
Right, I'm pretty sure that there's a way to do it (by using the `maybe_typeck_results` pattern I mentioned back then on Zulip). After all, the cleanup part is all about getting rid of `{visit,handle}_pat` for me personally.

I'm gonna be afk in a sec, so I can't explain the `maybe_typeck_results` approach rn and I don't know if you can decipher my past rambling on Zulip. Well, you can check out rustc since it uses the "`maybe_typeck_results` pattern" quite frequently.

Reply:
It's fine. Gonna give it a try and if I can't gonna let you do it since you seem to have a clear implementation in mind.

Reply:
I'll experiment later. Ofc, you can always try reading through `hir::intravisit` and see which methods don't get called in this PR which do get called on master, maybe there's a more obvious 'bug'.

Reply:
I uncovered the `maybe_typeck_results` dark arcanes and fixed it. I wonder if there is a shorter way though, code still seems longer than it needs to be...

Reply:
I'll check that soon. In any case, since this fixes things / adds support for more path types, I'd say it'd be fine even if we couldn't simplify it.

Also, with this method rustdoc should now be able to resolve **fully-qualified paths** (`<$Type as $TraitRef>::$assoc`), too, and that in *all* positions (expr, pat, type). On master, we don't support those at all iirc. However,  I haven't double-checked if your impl is sufficient or if you still need to hook up some things. Could you add tests for them if this PR makes them work?